packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -83,43 +84,52 @@ function SelectionTool(props: Props) {
       const { target, pointerId } = sourceEvent;
       (target as Element).setPointerCapture(pointerId);
 
-      setStartEvt(evt);
-
-      if (onSelectionStart) {
-        onSelectionStart(evt);
-      }
+      startEvtRef.current = evt;
+      onSelectionStart?.(evt);
     },
     [onSelectionStart, shouldInteract]
   );
 
   const onPointerMove = useCallback(
     (evt: CanvasEvent<PointerEvent>) => {
-      if (startEvt) {
-        setSelection(computeSelection(evt));
+      if (!startEvtRef.current) {
+        return;
+      }
+      const newSelection = transformSelection(computeSelection(evt));
+
+      if (isModifierKeyPressed) {
```

Comment:
I think we already discussed this but why the condition is `isModifierKeyPressed` rather than `shouldInteract` ?

Reply:
Instinctively, I'd say that we don't need to because the interaction was allowed to start. Maybe also the buttons in the event aren't right in `pointermove` or something like that? I wonder if the question should rather be if the condition in `onPointerUp` should be `isModifierKeyPressed` instead of `shouldInteract`...  😂

Reply:
Yeah, that rings a bell. Perhaps we should rename `shouldInteract` into `shouldInteractionStart` in the future.

Anyway, fine for me.